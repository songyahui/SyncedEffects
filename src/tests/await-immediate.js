"use hopscript"

var hh = require("hiphop");

var inSig = {accessibility: hh.IN};
var outSig = {accessibility: hh.OUT};

var prg =
    <hh.module I=${inSig} O=${outSig}>
      <hh.loop>
	<hh.sequence>
	  <hh.await I immediate />
	  <hh.emit O />
	  <hh.pause/>
	</hh.sequence>
      </hh.loop>
    </hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "awaitimmediate");
