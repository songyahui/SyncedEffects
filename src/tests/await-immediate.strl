module awaitimmediate:

input I;
output O;

loop
	await immediate I;
	emit O;
	pause;
end loop

end module