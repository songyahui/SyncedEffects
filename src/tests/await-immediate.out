awaitimmediate> I;
--- Output: O
awaitimmediate> ;
--- Output:
awaitimmediate> ;
--- Output:
awaitimmediate> I;
--- Output: O
awaitimmediate> I;
--- Output: O
awaitimmediate> ;
--- Output:
